### PR TITLE
10965 move animgraph parameter props to inspector

### LINKED ISSUE #10965: Move anim graph parameter properties to inspector
[Migrated from JIRA LYN-11489]

**Description**
The properties of a parameter that are currently shown inside the "Edit Parameter" window shall be moved into the Inspector window.

**Old workflow:**
Anim graph parameters are shown and edited in the "Parameters" window on the bottom right. They are listed by their name and have a widget to control the current value.

image

When selecting a parameter and then pressing the gears icon, the properties of the parameter appear in a pop-up window named "Edit Parameter".

image

The content from the window shown in the image above shall be moved to the "Inspector" window and should be shown when selecting a parameter. The gears icon shall be removed from the "Parameters" window. Also, the context menu option named "Edit" that can be opened on a parameter shall be removed.

**New way / the change:**
The whole content of the "Edit Parameter" window shall be moved into the "Inspector" window, including the "Apply" and "Cancel" buttons. The source code for the "Edit Parameter" window can be found in the ParameterCreateEditDialog.h/.cpp files. This QDialog needs to be changed into a QWidget so that it can be sent to the Inspector window.

In case a new single parameter gets selected, the new widget needs to be sent to the "Inspector" window by calling
`EMStudio::InspectorRequestBus::Broadcast(&EMStudio::InspectorRequestBus::Events::Update,...)`
Please see other instances like for example the `AttributesWindow `that also communicates with the Inspector as reference.

The result should look something like this:
image

**Acceptance Criteria:**
* The gears icon is not present in the "Parameters" window anymore.
* The context menu that can be opened on a selected parameter in the "Parameters" window is not present anymore.
* The content of the old "Edit Parameter" window is shown in the "Inspector" window when clicking/selecting a parameter. When clicking on e.g. an anim graph node in the "Anim Graph" window, the properties of the node are shown.
* When multiple parameters are shown, the Inspector window will go back into a no selection state as we currently can't display the properties of multiple parameters.

### PR DESCRIPTION
**Pull request by MiKom**
## What does this PR do?

This moves the parameter properties edit/creation window into the inspector.
I's a re-submission of PR [!11702](https://github.com/o3de/o3de/pull/11702) that I had to take over. See that PR for technical discussion. I only added two commits that fixed minor issues and rebased the commit on top of the `development` branch tip because otherwise, the PR was accepted.

https://user-images.githubusercontent.com/104767/196743764-10518a2a-2058-45d2-91f1-c063b144edf3.mp4

## How was this PR tested?

Manually according to requirements. Also EMotionFX unit test suite was updated to test the functionality.

Fixes #10965

### CODE REVIEW

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/ParameterCreateEditWidget.cpp [hunk 1]
```diff
@@ -25,51 +26,36 @@
...
-        }
-        else
-        {
-            m_parameter.reset(MCore::ReflectionSerializer::Clone(editParameter));
-            setWindowTitle("Edit Parameter");
-            m_createButton = new QPushButton("Apply", this);
-            m_originalName = editParameter->GetName();
-        }
-        m_createButton->setObjectName("EMFX.ParameterCreateEditDialog.CreateApplyButton");
+        m_createButton = new QPushButton("Create", this);
```
Comment (cgalvan) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/ParameterCreateEditWidget.cpp:40]: Nit: Any user-facing strings should be wrapped in `tr` so we can handle translation in the future. (Here and several other spots in the PR)

```suggestion
        m_createButton = new QPushButton(tr("Create"), this);
```

**Review by cgalvan — APPROVED** [review]

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/ParameterCreateEditWidget.cpp (hunk 1 above)
Comment (MiKom) [reply Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/ParameterCreateEditWidget.cpp]: Agreed

**Comment by amzn-rhhong** [comment]
Fantastic, thanks for providing videos. Have you done more adhock test with adding/removing the parameters?(while the inspector widget still opens)

**Review by amzn-rhhong — APPROVED** [review]

**Comment by MiKom** [comment]
> Fantastic, thanks for providing videos. Have you done more adhock test with adding/removing the parameters?(while the inspector widget still opens)

Hey, I did some more clicking inspired by your comment around and I managed to make it crash in one situation. I'll fix it. For now I'll make it a draft again.

**Comment by hultonha** [comment]
> > > Fantastic, thanks for providing videos. Have you done more adhock test with adding/removing the parameters?(while the inspector widget still opens)
> > 
> > 
> > Hey, I did some more clicking around inspired by your comment and I managed to make it crash in one situation. I'll fix it. For now I'll make it a draft again until the bug is fixed.
> 
> Added a commit that fixes the bug I found.

That's great to hear, I can kick off a new build now 👍

**Review by hultonha — APPROVED** [review]

**Comment by MiKom** [comment]
Seems that the CI errors are due to some problem with the build process itself. Like multiple compiler processes writing to the same files for some reason.

**Comment by hultonha** [comment]
> Seems that the CI errors are due to some problem with the build process itself. Like multiple compiler processes writing to the same files for some reason.
> 
> https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-12694/3/pipeline

Hmm okay I see, I'll try another now and see how it goes. If that fails it might be worth merging development into your branch again. Thanks!

**Pull request merged**